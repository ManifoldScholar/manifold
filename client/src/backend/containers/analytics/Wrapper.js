import React from "react";
import PropTypes from "prop-types";
import { useTranslation } from "react-i18next";
import { childRoutes } from "helpers/router";
import HeadContent from "global/components/HeadContent";

export default function AnalyticsWrapperContainer({ route }) {
  const { t } = useTranslation();

  return (
    <>
      <HeadContent
        title={`${t("titles.analytics")} | ${t("common.admin")}`}
        appendDefaultTitle
      />
      <main id="skip-to-main" tabIndex={-1}>
        <h1 className="screen-reader-text">{t("dashboard.title")}</h1>
        <section>
          <div className="container">{childRoutes(route)}</div>
        </section>
      </main>
    </>
  );
}

AnalyticsWrapperContainer.propTypes = {
  route: PropTypes.object
};
