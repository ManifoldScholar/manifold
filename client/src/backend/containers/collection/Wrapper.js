import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import connectAndFetch from "utils/connectAndFetch";
import Layout from "backend/components/layout";
import Dialog from "backend/components/dialog";
import Navigation from "backend/components/navigation";
import Utility from "global/components/utility";
import { entityStoreActions, notificationActions } from "actions";
import { select } from "utils/entityUtils";
import { collectionsAPI, requests } from "api";
import lh from "helpers/linkHandler";
import { childRoutes, RedirectToFirstMatch } from "helpers/router";
import navigation from "helpers/router/navigation";

import Authorize from "hoc/authorize";

const { request, flush } = entityStoreActions;

export class CollectionWrapperContainer extends PureComponent {
  static mapStateToProps = (state, ownPropsIgnored) => {
    return {
      collection: select(requests.beCollection, state.entityStore)
    };
  };

  static displayName = "Collection.Wrapper";

  static propTypes = {
    collection: PropTypes.object,
    dispatch: PropTypes.func,
    match: PropTypes.object,
    route: PropTypes.object,
    history: PropTypes.object
  };

  constructor(props) {
    super(props);
    this.state = {
      confirmation: null
    };
  }

  componentDidMount() {
    this.fetchCollection();
  }

  componentWillUnmount() {
    this.props.dispatch(flush(requests.beCollection));
  }

  fetchCollection = () => {
    const call = collectionsAPI.show(this.props.match.params.id);
    const collectionRequest = request(call, requests.beCollection);
    this.props.dispatch(collectionRequest);
  };

  closeDialog() {
    this.setState({ confirmation: null });
  }

  doPreview = event => {
    event.preventDefault();
    const previewUrl = lh.link(
      "frontendResourceCollection",
      this.props.collection.relationships.project.attributes.slug,
      this.props.collection.attributes.slug
    );
    const win = window.open(previewUrl, "_blank");
    win.focus();
  };

  doDestroy = () => {
    const call = collectionsAPI.destroy(this.props.collection.id);
    const options = { removes: this.props.collection };
    const collectionRequest = request(
      call,
      requests.beCollectionDestroy,
      options
    );
    this.props.dispatch(collectionRequest).promise.then(() => {
      this.notifyDestroy();
      this.redirectToProjectCollections();
    });
  };

  redirectToProjectCollections() {
    const projectId = this.props.collection.relationships.project.id;
    const redirectUrl = lh.link("backendProjectResourceCollections", projectId);
    this.props.history.push(redirectUrl);
  }

  notifyDestroy() {
    const notification = {
      level: 0,
      id: `COLLECTION_DESTROYED_${this.props.collection.id}`,
      heading: "The collection has been destroyed.",
      body: `${
        this.props.collection.attributes.title
      } has passed into the endless night.`,
      expiration: 5000
    };
    this.props.dispatch(notificationActions.addNotification(notification));
  }

  handleCollectionDestroy = event => {
    const heading = "Are you sure you want to delete this collection?";
    const message = "This action cannot be undone.";
    new Promise((resolve, reject) => {
      this.setState({
        confirmation: { resolve, reject, heading, message }
      });
    }).then(
      () => {
        this.doDestroy(event);
        this.closeDialog();
      },
      () => {
        this.closeDialog();
      }
    );
  };

  renderUtility() {
    return (
      <div>
        <button onClick={this.doPreview} className="button-bare-primary">
          <i className="manicon manicon-eye-outline" aria-hidden="true" />
          Preview{" "}
        </button>
        <button
          onClick={this.handleCollectionDestroy}
          className="button-bare-primary"
        >
          <i className="manicon manicon-trashcan" aria-hidden="true" />
          Delete
        </button>
      </div>
    );
  }

  renderRoutes() {
    const { collection } = this.props;
    return childRoutes(this.props.route, { childProps: { collection } });
  }

  render() {
    /* eslint-disable no-unused-vars */
    const { collection, match } = this.props;
    /* eslint-enable no-unused-vars */
    if (!collection) return null;
    const skipId = "skip-to-collection-panel";
    const secondaryLinks = navigation.collection(collection);

    return (
      <div>
        <Authorize
          entity={collection}
          failureFatalError={{
            body: "You are not allowed to edit this collection."
          }}
          ability="update"
        >
          <RedirectToFirstMatch
            from={lh.link("backendCollection", collection.id)}
            candidates={secondaryLinks}
          />
          {this.state.confirmation ? (
            <Dialog.Confirm {...this.state.confirmation} />
          ) : null}
          <Navigation.DetailHeader
            type="collection"
            backUrl={lh.link(
              "backendProjectResourceCollections",
              collection.relationships.project.id
            )}
            backLabel={
              collection.relationships.project.attributes.titlePlaintext
            }
            utility={this.renderUtility()}
            title={collection.attributes.title}
            secondaryLinks={secondaryLinks}
          />
          <Layout.BackendPanel
            sidebar={<Navigation.Secondary links={secondaryLinks} panel />}
          >
            <Utility.SkipLink skipId={skipId} />
            <div id={skipId}>{this.renderRoutes()}</div>
          </Layout.BackendPanel>
        </Authorize>
      </div>
    );
  }
}

export default connectAndFetch(CollectionWrapperContainer);