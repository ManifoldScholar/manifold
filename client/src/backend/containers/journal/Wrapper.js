import React, { useCallback } from "react";
import Layout from "backend/components/layout";
import withConfirmation from "hoc/withConfirmation";
import { journalsAPI } from "api";
import { childRoutes } from "helpers/router";
import lh from "helpers/linkHandler";
import navigation from "helpers/router/navigation";
import Authorize from "hoc/Authorize";
import {
  useFetch,
  useApiCallback,
  useNotification,
  useRedirectToFirstMatch
} from "hooks";
import { useTranslation } from "react-i18next";
import HeadContent from "global/components/HeadContent";
import PageHeader from "backend/components/layout/PageHeader";
import { RegisterBreadcrumbs } from "global/components/atomic/Breadcrumbs";

function JournalWrapper({ match, route, history, confirm, location }) {
  const { t } = useTranslation();
  const { data: journal, refresh } = useFetch({
    request: [journalsAPI.show, match.params.id]
  });
  const destroy = useApiCallback(journalsAPI.destroy, { removes: journal });

  const notifyDestroy = useNotification(j => ({
    level: 0,
    id: `JOURNAL_DESTROYED_${j.id}`,
    heading: t("notifications.journal_delete"),
    body: t("notifications.delete_entity_body", {
      title: j?.attributes?.title
    }),
    expiration: 5000
  }));

  const destroyAndRedirect = useCallback(() => {
    const redirect = () => history.push(lh.link("backendJournals"));
    destroy(journal.id).then(
      () => {
        notifyDestroy(journal);
        redirect();
      },
      () => redirect()
    );
  }, [destroy, history, journal, notifyDestroy]);

  const handleJournalDestroy = useCallback(() => {
    const heading = t("modals.delete_journal");
    const message = t("modals.confirm_body");
    confirm(heading, message, destroyAndRedirect);
  }, [destroyAndRedirect, confirm, t]);

  const utility = [
    {
      label: "actions.view",
      route: "frontendJournalDetail",
      slug: journal?.attributes.slug,
      icon: "eyeOpen32"
    },
    {
      label: "actions.delete",
      authorize: "delete",
      icon: "delete32",
      onClick: handleJournalDestroy
    }
  ];

  const renderRoutes = () => {
    return childRoutes(route, {
      childProps: { refresh, journal }
    });
  };

  useRedirectToFirstMatch({
    route: "backendJournal",
    id: journal?.id,
    slug: journal?.attributes.slug,
    candidates: journal ? navigation.journal(journal) : []
  });

  if (!journal) return null;

  const subpage = location.pathname.split("/")[4]?.replace("-", "_");

  const breadcrumbs = [
    {
      to: lh.link("backendJournals"),
      label: t("glossary.journal_title_case_other")
    },
    {
      to: lh.link("backendJournals", journal.id),
      label: journal.attributes.titlePlaintext
    }
  ];

  return (
    <div>
      <Authorize
        entity={journal}
        failureFatalError={{
          detail: t("journals.unauthorized_edit")
        }}
        ability={["read"]}
      >
        {subpage && (
          <HeadContent
            title={`${t(`titles.${subpage}`)} | ${
              journal.attributes.titlePlaintext
            } | ${t("common.admin")}`}
            appendDefaultTitle
          />
        )}
        <RegisterBreadcrumbs breadcrumbs={breadcrumbs ?? []} />
        <PageHeader
          type="journal"
          title={journal.attributes.titleFormatted}
          subtitle={journal.attributes.subtitle}
          actions={utility}
          secondaryLinks={navigation.journal(journal)}
          issues={journal.attributes.issuesNav}
        />
        <Layout.BackendPanel
          sidebar={
            <Layout.SecondaryNav
              links={navigation.journal(journal)}
              panel
              ariaLabel={t("journals.settings")}
            />
          }
        >
          <div>{renderRoutes()}</div>
        </Layout.BackendPanel>
      </Authorize>
    </div>
  );
}

export default withConfirmation(JournalWrapper);
