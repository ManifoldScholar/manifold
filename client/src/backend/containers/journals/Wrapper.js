import React from "react";
import PropTypes from "prop-types";
import { useTranslation } from "react-i18next";
import { childRoutes } from "helpers/router";
import Authorize from "hoc/Authorize";

export default function JournalsWrapper({ route }) {
  const { t } = useTranslation();
  return (
    <Authorize
      ability="update"
      entity={["journal"]}
      failureFatalError={{
        body: t("journals.unauthorized")
      }}
    >
      <div>
        <main id="skip-to-main" tabIndex={-1} className="backend-detail">
          {childRoutes(route)}
        </main>
      </div>
    </Authorize>
  );
}

JournalsWrapper.propTypes = {
  route: PropTypes.object
};
