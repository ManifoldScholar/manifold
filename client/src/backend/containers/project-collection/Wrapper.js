import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { withTranslation } from "react-i18next";
import connectAndFetch from "utils/connectAndFetch";
import { collectionProjectsAPI, projectCollectionsAPI, requests } from "api";
import { entityStoreActions } from "actions";
import { select } from "utils/entityUtils";
import EntityCollectionPlaceholder from "global/components/entity/CollectionPlaceholder";
import ProjectCollection from "backend/components/project-collection";
import New from "./New";
import Drawer from "global/components/drawer";
import { childRoutes } from "helpers/router";
import size from "lodash/size";
import lh from "helpers/linkHandler";
import classnames from "classnames";
import HeadContent from "global/components/HeadContent";
import withConfirmation from "hoc/withConfirmation";
import { RegisterBreadcrumbs } from "global/components/atomic/Breadcrumbs";
import { fluidScale } from "theme/styles/mixins";

import Authorize from "hoc/Authorize";

const { request, flush } = entityStoreActions;

export class ProjectCollectionWrapperContainer extends PureComponent {
  static mapStateToProps = (state, ownPropsIgnored) => {
    return {
      projectCollections: select(
        requests.beProjectCollections,
        state.entityStore
      ),
      projectCollection: select(
        requests.beProjectCollection,
        state.entityStore
      ),
      collectionProjects: select(
        requests.beCollectionProjects,
        state.entityStore
      )
    };
  };

  static displayName = "ProjectCollection.Wrapper";

  static propTypes = {
    projectCollections: PropTypes.array,
    collectionProjects: PropTypes.array,
    refresh: PropTypes.func,
    dispatch: PropTypes.func,
    history: PropTypes.object,
    match: PropTypes.object,
    route: PropTypes.object,
    confirm: PropTypes.func,
    t: PropTypes.func
  };

  constructor(props) {
    super(props);
    this.state = { showNew: false, newFormDirty: false };
  }

  componentDidMount() {
    this.fetchProjectCollections();
    this.fetchProjectCollection();
    this.fetchCollectionProjects();
  }

  componentWillUnmount() {
    this.props.dispatch(flush(requests.beProjectCollection));
    this.props.dispatch(flush(requests.beProjectCollections));
    this.props.dispatch(flush(requests.beProjects));
    this.props.dispatch(flush(requests.beCollectionProjects));
  }

  componentDidUpdate(prevProps) {
    if (prevProps.match.params.id !== this.props.match.params.id) {
      this.fetchProjectCollection();
      this.fetchCollectionProjects();
    }
  }

  fetchProjectCollections = () => {
    const call = projectCollectionsAPI.index(
      { order: "position ASC" },
      { number: 1, size: 100 }
    );
    const { promise } = this.props.dispatch(
      request(call, requests.beProjectCollections)
    );
    return promise;
  };

  fetchProjectCollection = (page = 1, perPage = 12) => {
    const id = this.props.match.params.id;
    if (!id) return Promise.resolve();

    const pageParams = { number: page, size: perPage };
    const pagination = { collectionProjects: pageParams };

    const call = projectCollectionsAPI.show(id, pagination);
    const { promise } = this.props.dispatch(
      request(call, requests.beProjectCollection)
    );
    return promise;
  };

  fetchCollectionProjects = () => {
    const id = this.props.match.params.id;
    if (!id) return Promise.resolve();
    const call = collectionProjectsAPI.index(id);
    const { promise } = this.props.dispatch(
      request(call, requests.beCollectionProjects)
    );
    return promise;
  };

  destroyProjectCollection = (afterDestroy = this.returnToList) => {
    const projectCollection = this.props.projectCollection;
    const call = projectCollectionsAPI.destroy(projectCollection.id);
    const options = { removes: projectCollection };
    const destroyRequest = request(
      call,
      requests.beProjectCollectionDestroy,
      options
    );
    const { promise } = this.props.dispatch(destroyRequest);
    promise.then(afterDestroy);
    return promise;
  };

  returnToList = () => {
    return this.props.history.push(lh.link("backendProjectCollections"));
  };

  buildUpdateProjectCollection = (id, changes) => {
    return projectCollectionsAPI.update(id, changes);
  };

  buildCreateProjectCollection = model => {
    return projectCollectionsAPI.create(model);
  };

  updateCollectionProject = (id, changes, options = {}) => {
    const projectCollection = this.props.projectCollection;
    const call = projectCollectionsAPI.updateCollectionProject(
      projectCollection.id,
      id,
      changes
    );
    const projectCollectionRequest = request(
      call,
      requests.beCollectionProjectUpdate,
      options
    );
    const { promise } = this.props.dispatch(projectCollectionRequest);
    return promise;
  };

  updateProjectCollection = (
    changes,
    options = {},
    optionalProjectCollection = null
  ) => {
    const projectCollection =
      optionalProjectCollection || this.props.projectCollection;
    const config = this.buildUpdateProjectCollection(
      projectCollection.id,
      changes
    );
    const updateRequest = request(
      config,
      requests.beProjectCollectionUpdate,
      options
    );
    const { promise } = this.props.dispatch(updateRequest);
    promise.then(this.fetchCollectionProjects);
    return promise;
  };

  handleCollectionOrderChange = result => {
    const changes = { attributes: { position: result.position } };
    this.updateProjectCollection(
      changes,
      { noTouch: true },
      { id: result.id }
    ).then(this.fetchProjectCollections);
  };

  handleCollectionSelect = collection => {
    const url = lh.link("backendProjectCollection", collection.id);
    this.props.history.push(url);
  };

  handleShowNew = event => {
    if (event) event.preventDefault();
    this.setState({ showNew: true });
  };

  handleHideNew = event => {
    if (event) event.preventDefault();
    if (this.state.newFormDirty) {
      const heading = this.props.t("messages.confirm");
      const message = this.props.t("messages.unsaved_changes");
      return this.props.confirm(heading, message, () =>
        this.setState({ showNew: false })
      );
    }
    this.setState({ showNew: false });
  };

  handleNewSuccess = projectCollection => {
    this.fetchProjectCollections();
    const path = lh.link("backendProjectCollection", projectCollection.id);
    this.props.history.push(path);
    this.handleHideNew();
  };

  handleToggleVisibility = (projectCollection, visible) => {
    const changes = { attributes: { visible } };
    this.updateProjectCollection(changes, {}, projectCollection);
  };

  get childProps() {
    return {
      projectCollection: this.props.projectCollection,
      projectCollections: this.props.projectCollections,
      collectionProjects: this.props.collectionProjects,
      destroyProjectCollection: this.destroyProjectCollection,
      updateProjectCollection: this.updateProjectCollection,
      updateCollectionProject: this.updateCollectionProject,
      buildUpdateProjectCollection: this.buildUpdateProjectCollection,
      buildCreateProjectCollection: this.buildCreateProjectCollection,
      refreshProjectCollection: this.fetchProjectCollection,
      refreshProjectCollections: this.fetchProjectCollections,
      refreshCollectionProjects: this.fetchCollectionProjects,
      drawerProps: {
        closeUrl: lh.link("backendProjectCollections")
      }
    };
  }

  renderChildRoutes() {
    return childRoutes(this.props.route, {
      childProps: this.childProps
    });
  }

  get hasProjectCollections() {
    return !this.noProjectCollections;
  }

  get noProjectCollections() {
    return size(this.props.projectCollections) === 0;
  }

  render() {
    const t = this.props.t;
    const projectCollection = this.props.projectCollection;
    const projectCollections = this.props.projectCollections;
    if (!projectCollections) return null;

    const wrapperClasses = classnames("project-collections", {
      "active-collection": this.props.match.params.id || this.state.showNew,
      empty: this.noProjectCollections
    });

    const collectionForHeader = this.props.match.params.id
      ? projectCollection
      : null;
    const pageTitle = collectionForHeader
      ? collectionForHeader.attributes.title
      : t("titles.project_collections");

    const breadcrumbs = collectionForHeader
      ? [
          { to: null, label: t("glossary.project_title_case_other") },
          {
            to: lh.link("backendProjectCollections"),
            label: t("glossary.project_collection_title_case_other")
          },
          {
            to: lh.link("backendProjectCollections", projectCollection.id),
            label: projectCollection.attributes.title
          }
        ]
      : [
          { to: null, label: t("glossary.project_title_case_other") },
          {
            to: lh.link("backendProjectCollections"),
            label: t("glossary.project_collection_title_case_other")
          }
        ];

    return (
      <Authorize
        ability="update"
        entity={["projectCollection"]}
        failureFatalError={{
          detail: this.props.t("project_collections.unauthorized")
        }}
      >
        <HeadContent
          title={`${pageTitle} | ${t("common.admin")}`}
          appendDefaultTitle
        />
        <RegisterBreadcrumbs breadcrumbs={breadcrumbs} />
        <section className={wrapperClasses}>
          <div className="backend-panel">
            <div
              className="container"
              style={{ marginBlockStart: fluidScale("30px", "20px") }}
            >
              {this.hasProjectCollections && (
                <ProjectCollection.List
                  projectCollection={projectCollection}
                  projectCollections={projectCollections}
                  onCollectionSelect={this.handleCollectionSelect}
                  onCollectionOrderChange={this.handleCollectionOrderChange}
                  onToggleVisibility={this.handleToggleVisibility}
                  onShowNew={this.handleShowNew}
                />
              )}
              <div className="panel">
                {this.hasProjectCollections && (
                  <ProjectCollection.Header
                    projectCollection={collectionForHeader}
                  />
                )}
                <Drawer.Wrapper
                  closeCallback={this.handleHideNew}
                  open={this.state.showNew}
                  context="backend"
                  size="flexible"
                  padding="large"
                  lockScroll="always"
                >
                  <New
                    successHandler={this.handleNewSuccess}
                    setDirty={val => this.setState({ newFormDirty: val })}
                    {...this.childProps}
                  />
                </Drawer.Wrapper>
                {this.noProjectCollections ? (
                  <EntityCollectionPlaceholder.ProjectCollectionsBackend
                    onClick={this.handleShowNew}
                  />
                ) : (
                  this.renderChildRoutes()
                )}
              </div>
            </div>
          </div>
        </section>
      </Authorize>
    );
  }
}

export default withTranslation()(
  withConfirmation(connectAndFetch(ProjectCollectionWrapperContainer))
);
