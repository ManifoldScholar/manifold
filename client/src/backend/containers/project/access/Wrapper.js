import React, { Component } from "react";
import PropTypes from "prop-types";
import { withTranslation } from "react-i18next";
import { childRoutes } from "helpers/router";
import lh from "helpers/linkHandler";
import PermissionsContainer from "backend/containers/permission";
import EntitlementsContainer from "backend/containers/entitlements";
import Authorize from "hoc/Authorize";
import Authorization from "helpers/authorization";
import Layout from "backend/components/layout";
import FormContainer from "global/containers/form";
import Form from "global/components/form";
import Hero from "backend/components/hero";
import withSettings from "hoc/withSettings";
import connectAndFetch from "utils/connectAndFetch";

class ProjectAccessWrapper extends Component {
  static mapStateToProps = state => {
    return {
      authentication: state.authentication
    };
  };

  static displayName = "Project.Access.Wrapper";

  static propTypes = {
    project: PropTypes.object,
    route: PropTypes.object,
    history: PropTypes.object,
    match: PropTypes.object,
    authentication: PropTypes.object,
    t: PropTypes.func
  };

  get defaultIsRestricted() {
    return this.props.settings.attributes.general.restrictedAccess === true;
  }

  get defaultIsOpen() {
    return !this.defaultIsRestricted;
  }

  render() {
    const { project, updateProject, t, authentication } = this.props;
    if (!project) return null;

    const authorization = new Authorization();

    const canGrantPermissions = authorization.authorizeAbility({
      authentication,
      entity: project,
      ability: "managePermissions"
    });

    const closeUrl = lh.link("backendProjectAccess", project.id);
    return (
      <>
        <Authorize
          entity={project}
          ability="update"
          failureNotification
          failureRedirect={lh.link("backendProject", project.id)}
        >
          {canGrantPermissions && (
            <Layout.BackendPanel flush>
              <PermissionsContainer.List entity={project} />
            </Layout.BackendPanel>
          )}
          <Layout.BackendPanel flush={!canGrantPermissions}>
            <EntitlementsContainer.List
              entity={project}
              preList={
                <div style={{ marginBottom: 44, marginTop: 22 }}>
                  <Hero.Block
                    title={t("projects.forms.access.title")}
                    description={t("projects.forms.access.description")}
                    initialVisible={this.defaultIsRestricted}
                  >
                    <FormContainer.Form
                      style={{ paddingTop: 24, paddingBottom: 24 }}
                      model={project}
                      name="backend-project-update"
                      update={updateProject}
                      className="form-secondary"
                    >
                      {this.defaultIsOpen && (
                        <Form.Switch
                          label={t("projects.forms.access.restricted_label")}
                          name="attributes[restrictedAccess]"
                          wide
                        />
                      )}
                      {this.defaultIsRestricted && (
                        <Form.Switch
                          label={t("projects.forms.access.open_access_label")}
                          name="attributes[openAccess]"
                          wide
                        />
                      )}
                      <Form.TextInput
                        label={t("projects.forms.access.notice_header_label")}
                        name="attributes[restrictedAccessHeading]"
                        placeholder={t(
                          "projects.forms.access.notice_header_placeholder"
                        )}
                        wide
                      />
                      <Form.TextArea
                        label={t("projects.forms.access.notice_body_label")}
                        name="attributes[restrictedAccessBody]"
                        placeholder={t(
                          "projects.forms.access.notice_body_placeholder"
                        )}
                      />
                      <Form.Save text={t("projects.forms.access.save")} />
                    </FormContainer.Form>
                  </Hero.Block>
                </div>
              }
            />
          </Layout.BackendPanel>
        </Authorize>
        {childRoutes(this.props.route, {
          drawer: true,
          drawerProps: { closeUrl, lockScroll: "always" },
          childProps: {
            entity: project,
            closeUrl,
            history: this.props.history
          }
        })}
      </>
    );
  }
}

export default withTranslation()(
  withSettings(connectAndFetch(ProjectAccessWrapper))
);
