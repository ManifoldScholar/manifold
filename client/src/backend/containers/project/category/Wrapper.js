import React, { Component } from "react";
import PropTypes from "prop-types";
import { withRouter } from "react-router-dom";
import { childRoutes } from "helpers/router";

class ProjectCategoryWrapperContainer extends Component {
  static displayName = "Project.Category.Wrapper";

  static propTypes = {
    project: PropTypes.object,
    history: PropTypes.object,
    route: PropTypes.object
  };

  render() {
    return (
      <div>{childRoutes(this.props.route, { childProps: this.props })}</div>
    );
  }
}

export default withRouter(ProjectCategoryWrapperContainer);
