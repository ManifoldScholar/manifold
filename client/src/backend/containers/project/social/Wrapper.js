import React, { Component } from "react";
import PropTypes from "prop-types";
import TwitterQueries from "./TwitterQueries";
import { childRoutes } from "helpers/router";
import lh from "helpers/linkHandler";

import Authorize from "hoc/Authorize";

export default class ProjectSocialWrapperContainer extends Component {
  static displayName = "Project.Social.Wrapper";

  static propTypes = {
    project: PropTypes.object,
    route: PropTypes.object,
    history: PropTypes.object,
    match: PropTypes.object
  };

  render() {
    const project = this.props.project;
    const closeUrl = lh.link("backendProjectSocial", project.id);

    return (
      <Authorize
        entity={project}
        ability="manageSocials"
        failureNotification
        failureRedirect={lh.link("backendProject", project.id)}
      >
        <section>
          {childRoutes(this.props.route, {
            drawer: true,
            drawerProps: { closeUrl }
          })}
          <TwitterQueries project={project} match={this.props.match} />
        </section>
      </Authorize>
    );
  }
}
