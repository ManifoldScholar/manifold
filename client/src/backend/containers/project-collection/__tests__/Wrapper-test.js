import { ProjectCollectionWrapperContainer } from "../Wrapper";
import { BreadcrumbsProvider } from "global/components/atomic/Breadcrumbs";

describe("backend/containers/project-collection/Wrapper", () => {
  beforeEach(() => {
    testHelpers.startSession($dispatch, $user);
  });

  def("abilities", () => ({ update: true }));
  def("projectCollection", () =>
    factory("projectCollection", { attributes: { abilities } })
  );
  def("user", () => factory("user"));
  def("history", () => fixtures.history());
  def("route", () => fixtures.route());
  def("root", () => (
    <BreadcrumbsProvider>
      <ProjectCollectionWrapperContainer
        projectCollections={$projectCollections}
        collectionProjects={[]}
        refresh={jest.fn}
        dispatch={$dispatch}
        history={$history}
        match={{ params: {} }}
        route={$route}
        t={key => key}
      />
    </BreadcrumbsProvider>
  ));

  context("when there are project collections", () => {
    def("projectCollections", () => collectionFactory("projectCollection"));
    it("matches the snapshot when rendered", () => {
      expect(render($withApp($root)).html()).toMatchSnapshot();
    });
  });

  context("when there are no project collections", () => {
    def("projectCollections", () => []);
    it("matches the snapshot when rendered", () => {
      expect(render($withApp($root)).html()).toMatchSnapshot();
    });
  });
});
//
// describe("Backend.ProjectCollection.Wrapper container", () => {
//   const store = build.store();
//   const currentUser = build.entity.user("1");
//   store.dispatch({
//     type: "UPDATE_CURRENT_USER",
//     error: false,
//     payload: {
//       data: currentUser
//     }
//   });
//
//   it("renders correctly when no project collections", () => {
//     const component = renderer.create(
//       wrapWithRouter(
//         <Provider store={store}>
//           <ProjectCollectionWrapperContainer
//             projectCollections={[]}
//             collectionProjects={[]}
//             refresh={jest.fn}
//             dispatch={store.dispatch}
//             history={{}}
//             match={{ params: {} }}
//           />
//         </Provider>
//       )
//     );
//     const tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
//   });
//
//   it("renders correctly when no project collections", () => {
//     const component = renderer.create(
//       wrapWithRouter(
//         <Provider store={store}>
//           <ProjectCollectionWrapperContainer
//             projectCollections={[
//               build.entity.projectCollection("1"),
//               build.entity.projectCollection("2")
//             ]}
//             collectionProjects={[]}
//             refresh={jest.fn}
//             dispatch={store.dispatch}
//             history={{}}
//             match={{ params: { id: "1" } }}
//             route={{ routes: [] }}
//           />
//         </Provider>
//       )
//     );
//     const tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
//   });
// });
