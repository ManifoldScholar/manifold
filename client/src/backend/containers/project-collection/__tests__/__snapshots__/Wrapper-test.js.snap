// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`backend/containers/project-collection/Wrapper when there are no project collections matches the snapshot when rendered 1`] = `""`;

exports[`backend/containers/project-collection/Wrapper when there are project collections matches the snapshot when rendered 1`] = `""`;
